src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(object) {
+    if (object instanceof Herbivore && !object.hidden) {
+      object.health -= 50;
+    }
+
+    if (object.health <= 0) {
+      const deadAnimal = Animal.alive.findIndex((victim) => victim.health <= 0);
+
+      Animal.alive.splice(deadAnimal, 1);
+    }
```

Comment:
The logic for removing a dead animal from the `Animal.alive` array is correct, but it could be improved. Currently, it finds the index of the first animal with health less than or equal to zero and removes it. This works in this context, but if there are multiple animals with health less than or equal to zero, it might not remove the correct one. Consider using `findIndex` with a more specific condition or directly using `splice` with the current object.